osu.Game/Graphics/Containers/WaveContainer.cs

```diff
@@ -181,7 +184,16 @@ protected override void Update()
                 Height = Parent!.Parent!.DrawSize.Y * 1.5f;
             }
 
-            protected override void PopIn() => Schedule(() => this.MoveToY(FinalPosition, APPEAR_DURATION, easing_show));
+            protected override void PopIn()
+            {
+                Schedule(() =>
+                {
+                    // In UserProfileOverlay and BeatmapSetOverlay, we call Hide() then Show() in sequence.
+                    // Using FinishTransforms ensures the complete popIn animation plays correctly.
+                    FinishTransforms();
```

Comment:
This will break animations in other cases (like toggling an overlay via hotkey F8 or F9).